src/Agent.Plugins/GitSourceProvider.cs

```diff
@@ -228,7 +228,9 @@ public abstract class GitSourceProvider : ISourceProvider
             {
                 // the endpoint should either be the SystemVssConnection (id = guild.empty, name = SystemVssConnection)
                 // or a real service endpoint to external service which has a real id
-                endpoint = executionContext.Endpoints.Single(x => (repository.Endpoint.Id != Guid.Empty && x.Id == repository.Endpoint.Id) || (repository.Endpoint.Id == Guid.Empty && string.Equals(x.Name, repository.Endpoint.Name, StringComparison.OrdinalIgnoreCase)));
+                endpoint = executionContext.Endpoints.Single(
+                    x => (repository.Endpoint.Id != Guid.Empty && x.Id == repository.Endpoint.Id) ||
+                    (repository.Endpoint.Id == Guid.Empty && string.Equals(x.Name, repository.Endpoint.Name.ToString(), StringComparison.OrdinalIgnoreCase)));
```

Comment:
why we need ToString()?

Reply:
The `ToString()` is needed to react to the changes Zac made where the `Endpoint.Name` is a `ExpressionValue<string>` instead of `string`. I synced up with Zac, and this is the preferred way to get the resolved name.